### PR TITLE
Postproc

### PR DESCRIPTION
**Pull request by jdegerickx**
- added majority voting as a potential postprocessing pipeline (default remains smooth_probabilities)
- bug fixing when downloading assets
- "-cleaned" now removed from filename downloaded asset
- various minor fixes and updates on date slider, get seasons and ui_map
- added product post processing after download (split into classification and confidence .tif with LUT and color maps)
- is_binary argument in crop type pipeline should depend on lookup_table?

### CODE REVIEW

File: notebooks/utils.py [hunk 1]
```diff
@@ -379,3 +371,137 @@ def train_classifier(inputs, targets):
...
+
+NODATAVALUE = {
+    "cropland": 255,
+    "croptype": 255,
+    "confidence": 255,
+}
+
+
+COLORMAP = {
+    "active-cropland": {
```
Comment (kvantricht) [inline notebooks/utils.py:392]: Why do we need this here?

File: src/worldcereal/job.py [hunk 2]
```diff
@@ -194,19 +201,28 @@ def generate_map(
...
-        filename_prefix=filename,
+        filename_prefix=product_type.value,
     )
 
+    # Get look-up tables
+    luts = {}
+    luts[WorldCerealProductType.CROPLAND.value] = load_model_lut(
+        cropland_parameters.classifier_parameters.classifier_url
+    )
+    if (
```
Comment (kvantricht) [inline src/worldcereal/job.py:218]: there's no way to avoid if statements here I guess?

File: src/worldcereal/openeo/postprocess.py [hunk 3]
```diff
@@ -32,6 +33,139 @@ def output_labels(self) -> list:
...
 
+    @classmethod
+    def majority_vote(
+        cls,
+        base_labels: xr.DataArray,
+        max_probabilities: xr.DataArray,
+        kernel_size: int,
+        conf_threshold: int,
+    ) -> xr.DataArray:
+        """
```
Comment (kvantricht) [inline src/worldcereal/openeo/postprocess.py:44]: can you reformat the docstring following the Numpy style format? Your autodocstring should be configured for that already I think?

File: src/worldcereal/openeo/postprocess.py [hunk 4]
```diff
@@ -32,6 +33,139 @@ def output_labels(self) -> list:
     def dependencies(self) -> list:
         return []
 
+    @classmethod
+    def majority_vote(
```
Comment (kvantricht) [inline src/worldcereal/openeo/postprocess.py:37]: This is tested for multiclass arrays?

File: src/worldcereal/openeo/postprocess.py [hunk 5]
```diff
@@ -64,6 +198,12 @@ def smooth_probabilities(
             )
             probabilities_vals[class_idx][excluded_mask] = 0
 
+        # Sum of probabilities should be 1
+        probabilities_vals = probabilities_vals / probabilities_vals.sum(axis=0)
+
+        # Cast back to uint16
+        probabilities_vals = np.round(probabilities_vals * 100.0).astype("uint16")
```
Comment (kvantricht) [inline src/worldcereal/openeo/postprocess.py:205]: can we do this in one go?
something like
np.round(`probabilities_vals / probabilities_vals.sum(axis=0) * 100.0).astype(np.uint16)`?

File: src/worldcereal/utils/map.py [hunk 6]
```diff
@@ -1,51 +1,14 @@
...
-        3: (247, 185, 29, 255),  # millet_sorghum
-        4: (186, 186, 186, 0),  # other_crop
-        5: (167, 245, 66, 255),  # rapeseed
-        6: (85, 218, 218, 255),  # soy
-        7: (245, 66, 111, 255),  # sunflower
-        8: (186, 113, 53, 255),  # wheat
-    },
-    "probabilities": get_probability_cmap(),
-}
+LOCALTILESERVER_PORT = 8889
```
Comment (kvantricht) [inline src/worldcereal/utils/map.py:11]: what do we still need this for? Does this local tile server stuff work? I thought we were not going to try to go this far.

File: notebooks/utils.py (hunk 1 above)
Comment (jdegerickx) [reply notebooks/utils.py:392]: is a remainder from phase I, will delete...

File: src/worldcereal/openeo/postprocess.py (hunk 4 above)
Comment (jdegerickx) [reply src/worldcereal/openeo/postprocess.py:37]: yes, tested it for multiclass crop type...

File: src/worldcereal/utils/map.py (hunk 6 above)
Comment (jdegerickx) [reply src/worldcereal/utils/map.py:11]: moving product visualization to utils.py...

File: src/worldcereal/job.py (hunk 2 above)
Comment (jdegerickx) [reply src/worldcereal/job.py:218]: not that I can think of no...

File: tests/worldcerealtests/test_postprocessing.py
Comment (kvantricht) [inline tests/worldcerealtests/test_postprocessing.py:1]: you could maybe add two or three validation tests for the parameters.
Like [here](https://github.com/WorldCereal/worldcereal-classification/blob/6dfa75824b7772a3a10f6d30c11331bbef834f49/tests/worldcerealtests/test_preprocessing.py#L20), you can check if an error is raised as expected, and if correct parameters just work.

**Review by kvantricht — APPROVED** [review]
One more comment on adding a test

**Pull request merged**